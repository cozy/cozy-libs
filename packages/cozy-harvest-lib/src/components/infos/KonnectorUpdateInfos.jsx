import React from 'react'
import PropTypes from 'prop-types'

import { translate } from 'cozy-ui/transpiled/react/I18n'
import Infos from 'cozy-ui/transpiled/react/Infos'

import KonnectorUpdateLinker from '../KonnectorUpdateLinker'

/**
 * Warns the user that a new version is available for the given konnector.
 * Offer a button to redirect to an app able to update a konnector
 * (typically the store).
 */
export const KonnectorUpdateInfos = props => {
  const { className, konnector, isBlocking, t } = props
  return (
    <Infos
      actionButton={
        <KonnectorUpdateLinker
          konnector={konnector}
          isBlocking={isBlocking}
          label={t('infos.konnectorUpdate.button.label')}
        />
      }
      className={className}
      isImportant={isBlocking}
      text={
        isBlocking
          ? t('infos.konnectorUpdate.body.blocking')
          : t('infos.konnectorUpdate.body.regular')
      }
      title={t('infos.konnectorUpdate.title')}
    />
  )
}

KonnectorUpdateInfos.propTypes = {
  konnector: PropTypes.object.isRequired,
  className: PropTypes.string,
  isBlocking: PropTypes.bool,
  t: PropTypes.func.isRequired
}

export default React.memo(translate()(KonnectorUpdateInfos))
