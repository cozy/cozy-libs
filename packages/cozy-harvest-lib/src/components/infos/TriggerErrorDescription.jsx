import PropTypes from 'prop-types'
import React from 'react'

import { useClient } from 'cozy-client'
import Typography from 'cozy-ui/transpiled/react/Typography'
import { useI18n } from 'cozy-ui/transpiled/react/providers/I18n'

import { getErrorLocale } from '../../helpers/konnectors'
import Markdown from '../Markdown'
import withKonnectorLocales from '../hoc/withKonnectorLocales'
import useSupportMail from '../hooks/useSupportMail'

const TriggerErrorDescription = ({ error, konnector, linkProps }) => {
  const { t } = useI18n()
  const client = useClient()
  const { fetchStatus, supportMail } = useSupportMail(client)

  if (fetchStatus !== 'loaded' || !supportMail) {
    return null
  }

  return (
    <Typography variant="body1" color="initial" component="div">
      <Markdown
        source={getErrorLocale(error, konnector, t, 'description', supportMail)}
        linkProps={linkProps}
      />
    </Typography>
  )
}

TriggerErrorDescription.propTypes = {
  error: PropTypes.object.isRequired,
  konnector: PropTypes.object.isRequired,
  linkProps: PropTypes.object
}

const TriggerErrorDescriptionWrapper = withKonnectorLocales(
  TriggerErrorDescription
)

const TriggerErrorDescriptionWrapperWrapper = ({ children, ...props }) => {
  const { lang } = useI18n()

  return (
    <TriggerErrorDescriptionWrapper lang={lang} {...props}>
      {children}
    </TriggerErrorDescriptionWrapper>
  )
}

export default TriggerErrorDescriptionWrapperWrapper
