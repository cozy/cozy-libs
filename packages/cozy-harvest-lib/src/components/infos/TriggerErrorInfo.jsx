import React, { PureComponent } from 'react'
import PropTypes from 'prop-types'

import { translate } from 'cozy-ui/transpiled/react/I18n'
import Infos from 'cozy-ui/transpiled/react/Infos'

import { getErrorLocale } from '../../helpers/konnectors'
import withKonnectorLocales from '../hoc/withKonnectorLocales'
import Markdown from '../Markdown'

/**
 * Component used to show an error related to an account or to a konnector job
 * Regular error are treated in a common way, but for KonnectorJobErrors
 * we can add a little bit of logic.
 *
 * This component is strongly related with the content of locales files and
 * deals mainly with translation concerns.
 */
export class TriggerErrorInfo extends PureComponent {
  render() {
    const { className, error, konnector, t } = this.props
    return (
      <Infos
        className={className}
        isImportant
        text={
          <Markdown
            source={getErrorLocale(error, konnector, t, 'description')}
          />
        }
        title={getErrorLocale(error, konnector, t, 'title')}
      />
    )
  }
}

TriggerErrorInfo.proptTypes = {
  error: PropTypes.object.isRequired,
  konnector: PropTypes.object.isRequired,
  t: PropTypes.func.isRequired
}

export default translate()(withKonnectorLocales(TriggerErrorInfo))
