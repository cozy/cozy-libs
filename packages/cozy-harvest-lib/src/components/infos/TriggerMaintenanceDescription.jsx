import get from 'lodash/get'
import PropTypes from 'prop-types'
import React from 'react'

import Typography from 'cozy-ui/transpiled/react/Typography'
import { useI18n } from 'cozy-ui/transpiled/react/providers/I18n'

import Markdown from '../Markdown'

const TriggerMaintenanceDescription = ({ maintenanceMessages }) => {
  const { lang } = useI18n()
  const longMessage = get(maintenanceMessages, [lang, 'long_message'])

  if (!longMessage) {
    return null
  }

  return (
    <Typography variant="body1" component="div">
      <Markdown source={longMessage} />
    </Typography>
  )
}

TriggerMaintenanceDescription.propTypes = {
  maintenanceMessages: PropTypes.shape({
    en: PropTypes.shape({
      long_message: PropTypes.string,
      short_message: PropTypes.string
    }),
    fr: PropTypes.shape({
      long_message: PropTypes.string,
      short_message: PropTypes.string
    })
  })
}

export default TriggerMaintenanceDescription
