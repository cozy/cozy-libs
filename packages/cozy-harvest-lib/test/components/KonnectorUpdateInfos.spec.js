/* eslint-env jest */
import React from 'react'
import { shallow } from 'enzyme'

import { KonnectorUpdateInfos } from 'components/infos/KonnectorUpdateInfos'

// Default props
const intents = {
  redirect: jest.fn()
}
const konnector = {
  slug: 'test-konnector'
}
const t = key => key

const props = {
  intents,
  konnector,
  t
}

describe('KonnectorUpdateInfos', () => {
  it('should render', () => {
    const component = shallow(<KonnectorUpdateInfos {...props} />).getElement()
    expect(component).toMatchSnapshot()
  })

  it('should render as blocking', () => {
    const component = shallow(
      <KonnectorUpdateInfos {...props} isBlocking={true} />
    ).getElement()
    expect(component).toMatchSnapshot()
  })
})
