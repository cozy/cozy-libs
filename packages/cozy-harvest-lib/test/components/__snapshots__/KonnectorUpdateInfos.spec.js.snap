// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`KonnectorUpdateInfos should render 1`] = `
<InfosMigration
  actionButton={
    <KonnectorUpdateLinker
      konnector={
        Object {
          "slug": "test-konnector",
        }
      }
      label="infos.konnectorUpdate.button.label"
    />
  }
  text="infos.konnectorUpdate.body.regular"
  title="infos.konnectorUpdate.title"
/>
`;

exports[`KonnectorUpdateInfos should render as blocking 1`] = `
<InfosMigration
  actionButton={
    <KonnectorUpdateLinker
      isBlocking={true}
      konnector={
        Object {
          "slug": "test-konnector",
        }
      }
      label="infos.konnectorUpdate.button.label"
    />
  }
  isImportant={true}
  text="infos.konnectorUpdate.body.blocking"
  title="infos.konnectorUpdate.title"
/>
`;
